crates/egui/src/widgets/plot/legend.rs

```diff
@@ -166,29 +169,30 @@ impl LegendWidget {
         rect: Rect,
         config: Legend,
         items: &[Box<dyn PlotItem>],
-        hidden_items: &ahash::HashSet<String>,
+        hidden_items: &mut ahash::HashSet<String>,
     ) -> Option<Self> {
         // Collect the legend entries. If multiple items have the same name, they share a
         // checkbox. If their colors don't match, we pick a neutral color for the checkbox.
-        let mut entries: BTreeMap<String, LegendEntry> = BTreeMap::new();
+        let mut entries: BTreeMap<String, LegendEntry> = BTreeMap::new(); 
+        let mut hidden_items_new: ahash::HashSet<String> = Default::default();
         items
             .iter()
             .filter(|item| !item.name().is_empty())
             .for_each(|item| {
-                entries
-                    .entry(item.name().to_owned())
-                    .and_modify(|entry| {
-                        if entry.color != item.color() {
-                            // Multiple items with different colors
-                            entry.color = Color32::TRANSPARENT;
-                        }
-                    })
-                    .or_insert_with(|| {
-                        let color = item.color();
-                        let checked = !hidden_items.contains(item.name());
-                        LegendEntry::new(color, checked)
-                    });
+                entries.entry(item.name().to_owned()).or_insert_with(|| {
+                    let color = item.color();
+                    let checked = !hidden_items.contains(item.name());
+                    let visible = item.visible();
+                    let hiddened= !(checked && visible || !checked && !visible);
+
+                    if hiddened {
```

Comment:
```suggestion
                    if checked != visible {
```

Reply:
…though I struggle to follow he logic

Reply:
another possible logic without default_visible but with visible checking if it's in hidden_items.
(_requires a PR to access hidden items, which seems necessary to know if the item is hidden in a future case.
and surely a specific logic for the visible._)
```
let visibility_points = !hidden_items.contains("Points");
plot_ui.points(points.name("Points").visible(visibility_points);
```


although with default_visible we set a state from the start, we don't have access to its modified state.
while with visible it will allow to have its state hidden or not dynamically.

I don't think they are the same thing all the same.
with default_visible we define a state from the start which is clear to me.
When to know if it is in hidden_items it is another problem.
I think I have a preference for default_visible to solve this feature.
But maybe an alternative and necessary PR to compare.